### ISSUE TITLE
Build breakage with P2P disabled

### ISSUE THREAD

**Issue by ebassi** [flatpak/flatpak#1279]
When building without P2P support, like GNOME Continuous does, and using GLib 2.55/master after [the `g_autolist` macros](https://bugzilla.gnome.org/show_bug.cgi?id=791342) landed, the build fails:

```
../common/flatpak-dir.h: In function ‘glib_listautoptr_cleanup_OstreeRepoFinderResult’:
../common/flatpak-dir.h:148:56: error: expected expression before ‘void’
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (OstreeRepoFinderResult, void)
                                                        ^
/usr/include/glib-2.0/glib/gmacros.h:462:116: note: in definition of macro ‘G_DEFINE_AUTOPTR_CLEANUP_FUNC’
   static inline void _GLIB_AUTOPTR_LIST_FUNC_NAME(TypeName) (GList **_l) { g_list_free_full (*_l, (GDestroyNotify) func); } \
                                                                                                                    ^
```

Because, of course, the second argument cannot be `void`.

**Comment by pwithnall** [comment]
Here’s a fix: https://github.com/flatpak/flatpak/pull/1293

**Comment by pwithnall** [comment]
Sorry about the problem. This issue was entirely a`void`able.

**Comment by mwleeds** [comment]
Fixed by b148160986cd6f829dd1dd186151dcada50c80fb
